mount/main.go

```diff
@@ -0,0 +1,70 @@
+//  Copyright 2021 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/cilium/pkg/mountinfo"
+
+	"golang.org/x/sys/unix"
+)
+
+func mountCgroup(cgroupRoot string) error {
+	cgroupRootStat, err := os.Stat(cgroupRoot)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if err = os.MkdirAll(cgroupRoot, 0755); err != nil {
+				return fmt.Errorf("unable to create cgroup mount directory: %s", err)
+			}
+		} else {
+			return fmt.Errorf("failed to stat the mount path %s: %s", cgroupRoot, err)
+		}
+	} else if !cgroupRootStat.IsDir() {
+		return fmt.Errorf("%s is a file which is not a directory", cgroupRoot)
+	}
```

Comment:
I think we can get rid of this code. We'll be creating the directory if not present from cilium agent daemonset.